compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -14116,6 +14116,203 @@ bool isSettingOp(uint8_t byteValue, TR::InstOpCode::Mnemonic SI_opcode)
       return false;
    }
 
+TR::Register *
+OMR::Z::TreeEvaluator::bitpermuteEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *value = node->getChild(0);
+   TR::Node *addr = node->getChild(1);
+   TR::Node *length = node->getChild(2);
+
+   auto valueReg = cg->evaluate(value);
+   auto addrReg = cg->evaluate(addr);
+
+   TR::Register *tmpReg = cg->allocateRegister(TR_GPR);
+
+   /// The bitpermuteEvaluator uses loop unrolling to do the vector bit permute operation required by the IL
+   /// However, the bitPermuteConstantUnrollThreshold constant defined below defines the cap for this technique 
+   /// (i.e. the maximum array size for which this implementation can be used). 
+   /// 
+   /// Currently it is set at 4. The reasons are as follows:
+   ///
+   /// 1. This technique scales linearly as the array size grows and is also expensive for the iCache.
+   /// 2. Additionally, on z14 and newer hardware, the vector implementation performs better than the loop unrolling
+   ///    technique for arrays of size > 4. Moreover, if a slightly modified version of the IL was available (better
+   ///    suited for BigEndian architectures) then the Vector implementation is on par or better than the loop
+   ///    unrolling technique for all array sizes.
+   ///
+   /// More info on the measurements is available here: https://github.com/eclipse/omr/pull/2330#issuecomment-378380538
+   /// The differences between the generated code for the different 
+   /// implementations can be seen here: https://github.com/eclipse/omr/pull/2330#issuecomment-378387516
```

Comment:
Doxygen does not do anything for local comments so the triple slash is unnecessary for this entire change. Can we revert to double slash?

Reply:
9410f34f7133d7a4cbcac91ef15bccd8856c884b